### PR TITLE
feat: Add new dashboard "Employees active"

### PR DESCRIPTION
**Pull request by gabrielThiffault**
# Objectives of the Pull Request ? 
Création d'un tableau de bord PBI et un Report builder représentant le nombre d'employés du CSS

# What is left out of the Pull Request ? 
Rien

# How to run the pull request ?
> Provide the code required to run the pull request. This is the code that will be used to review your pull request. **The provided code must work as-is. If a DBT error is raised while running the code, the PR will be rejected. The following code / placeholder is only provided as documentation / helper to get you started and you will need to adjust it.**

```bash
# Assuming you are in a folder containing both the core.data.store and the <cssXX>.data.store folder.
# Update the code
cd core.dashboards_store
git checkout feature/rh_emp_actif
git pull
# Might be required if you update either the poetry file or the lock file
# poetry shell && poetry lock && poetry install 
cd ../<cssXX>.data.tbe
git checkout feature/rh_emp_actif
git pull  
# Might be required if you add a new DBT dependency
 dbt deps
 dbt seed --full-refresh
# Run dbt 
dbt build --select +tag:emp_actif
```

# Checklist
> Please, read carefully each item before checking it. Your PR's review might be delayed otherwise.

* **Code** :
  * [X] The code I'm asking a review for is working. **I understand that my PR will be rejected as-is otherwise.**
  * [X] My code follows the conventions described in the `contribution guide`.
  * [X] My `models` / `seeds` / `adapters` are tested and documented in DBT 9through the `yaml` files.
  * I have a added a new **mandatory seed** / a new **mandatory adapter** : 
    * [] I have you populated the `tooling/nightly` project with my new seed / adapter.
    * [] An exemple of how to populate the adapter can be found in the `analysis` folder.
  * I have added a new **dashboard** : 
    * [X] I have updated the `core.dashboards_store/tooling/template/{{ cookiecutter.css_short_name + '.dashboards_store' }}/dbt_project.yml` file with my new dashboard. 
    * [] I have added a new `doc` page in the `core.dashboards_store/tooling/docs/content/` appropriate folder, describing what my dashboards does and how to configure it.
    * [x] The dashboard is saved with `.pbit` extension, **not** a `.pbix` extension, so it doesn't contains any data.
  * [X] I have called `sqfmt .` to format my code, and **made sure my code was still running as expected after having been reformated**.
* **Documentation** : 
  * [ ]  I have updated the documentation (README) accordingly to my changes.
  * [X]  The models I have added are documented in a `schema.yml` file.
* **Pull-request** : 
  * [x]  My merge message is prefixed with a `commitizen` key word so it looks something like `<feat|fix|chore|doc|refactor|perf|style>: foo bar`**
  * [X]  The code *provided to run the pull request* is working. **I understand that my PR will be rejected as-is otherwise.**
  * [X]  My pull request is documented. I have explained the needs for the PR, what was left out of the it and why.
  * [X]  I have carefully reviewd each changes made to a file and made sure the files included on the PR were actually added on purpose.
  * [X]  I have assignee at least **1** people to review my PR.

### CODE REVIEW

**Comment by sadqim** [comment]
@gabrielThiffault le permettant de rouler la PR ne fonctionne pas! peux tu revoir ce qui ne fonctionne pas stp

**Review by sadqim — CHANGES_REQUESTED** [review]
Compilation Error
  Model 'model.cssvt_tbe.dim_mapper_workplace' (models/marts/human_resources/dimensions/dim_mapper_workplace.sql) depends on a node named 'i_pai_tab_lieu_trav' which is disabled

**Comment by gabrielThiffault** [comment]
> rouler



> Compilation Error Model 'model.cssvt_tbe.dim_mapper_workplace' (models/marts/human_resources/dimensions/dim_mapper_workplace.sql) depends on a node named 'i_pai_tab_lieu_trav' which is disabled

Si possible de réessayer, svp

**Comment by sadqim** [comment]
@gabrielThiffault  ca ne fonctionne toujours pas! 
image

je sais pas si c'est la cause mais ton package ne pointe pas vers le bon répertoire
image

**Comment by sadqim** [comment]
> > @gabrielThiffault ca ne fonctionne toujours pas! image
> > je sais pas si c'est la cause mais ton package ne pointe pas vers le bon répertoire image
> 
> My bad, changement apporté

il vaut mieux nommé le répertoire core "core.dashboards_store" pour que ca fonctionne pour tout le monde

**Comment by sadqim** [comment]
> > > > @gabrielThiffault ca ne fonctionne toujours pas! image
> > > > je sais pas si c'est la cause mais ton package ne pointe pas vers le bon répertoire image
> > > 
> > > 
> > > My bad, changement apporté
> > 
> > 
> > il vaut mieux nommé le répertoire core "core.dashboards_store" pour que ca fonctionne pour tout le monde
> 
> Le nom du core a été changé dans le package.yml.
> 
> Mais avant ce changement, tout fonctionnait :
> 
> image

oui pour toi elle va rouler mais pour les autres qui vont rouler la PR sur ton répertoire cssvt il vont devoir faire un dbt deps(pour ca il faut que le package ait le même nom partout)

File: models/marts/human_resources/spines/spine_rh.sql [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{#
```
Comment (sadqimo) [inline models/marts/human_resources/spines/spine_rh.sql:1]: @gabrielThiffault à quoi sert cette table? elle n'est pas utiliser dans ce tdb?

File: models/marts/human_resources/features/schema.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+
+version: 2
+
+models:
+  - name: fact_tps_partial
```
Comment (sadqimo) [inline models/marts/human_resources/features/schema.yml:5]: @gabrielThiffault, je ne vois pas cette table. est ce qu'elle existe encore ?

File: models/marts/human_resources/features/schema.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    description: > 
+      This table list all the employees who are currently active as of today. We take the "stat_empl" seed, to see if the employee is active
+    tests:
+      - resolution:
+          combination_of_columns:
+            - matr
+            - corp_empl
+            - is_today_actif
+            - Type
+            - year_check
```
Comment (sadqimo) [inline models/marts/human_resources/features/schema.yml:44]: @gabrielThiffault il faut mettre a jour les schema.yml

File: models/marts/human_resources/features/active/schema.yml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  - name: emp_actif_fact_base
+    config:
+      tags: 
+        - emp_actif
+    description: >
+      Active table.
+      Map all employes which are active
+    columns:
+      - name: matr
+        description: The unique identifier of an employee.
```
Comment (sadqimo) [inline models/marts/human_resources/features/active/schema.yml:12]: @gabrielThiffault même commentaire, il faut le metre à jour le schema.yml

File: models/marts/human_resources/features/activity/fact_activity_current.sql [hunk 5]
```diff
@@ -31,7 +31,8 @@ with
         where school_year = {{ store.get_current_year() }} and is_main_job = 1  -- Retain metadata for the main job only
     )
 
-select src.matr, src.corp_empl, src.etat_empl, src.lieu_trav, src.stat_eng
+select
+    src.matr, src.corp_empl, src.etat_empl, src.lieu_trav, src.stat_eng, 1 as emp_actif
 from actives_etat_empl as src
 left join {{ ref("dim_employees") }} as dme on src.matr = dme.matr
 where last_pay_date > dateadd(week, -2, getdate())  -- Make sure the activte employes have been paid in the last two weeks
```
Comment (sadqimo) [inline models/marts/human_resources/features/activity/fact_activity_current.sql:38]: @gabrielThiffault je suis curieux :), est ce que tu as pu verifé si la volumétrie de cette table correspond à votre nombre total d'employés

**Comment by sadqimo** [comment]
@gabrielThiffault, peux tu m'explique stp la différence entre la table " emp_actif_fact_base", la "fact_activity_current" et la "fact_today_active"

File: models/dashboards/emp_actif/features/emp_actif_fact_emp_actif.sql [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        when stat.is_reg = 0 and act.type = 'A'
+        then 'Temporaire (paie automatique)'
+        when stat.is_reg = 0 and act.type = 'P'
+        then 'Temporaire (sur pièce)'
+        when stat.is_reg = 1
+        then 'Régulier'
+        else 'Autres'
+    end as 'type'
+from {{ ref("emp_actif_fact_base") }} as act
+left join {{ ref("dim_employees") }} as emp on act.matr = emp.matr
```
Comment (sadqimo) [inline models/dashboards/emp_actif/features/emp_actif_fact_emp_actif.sql:58]: @gabrielThiffault pour les joins ... est ce que tu es sur que c'est des left et non des inner ?

Comment (gabrielThiffault) [reply models/dashboards/emp_actif/features/emp_actif_fact_emp_actif.sql:58]: Ça sort le même nombre d'employés, mais oui t'as raison

File: models/marts/human_resources/features/activity/fact_activity_current.sql (hunk 5 above)
Comment (gabrielThiffault) [reply models/marts/human_resources/features/activity/fact_activity_current.sql:38]: Oui, sans les règles d'affaires nous arrivons à 2269 et le script lui donne 2231

File: models/marts/human_resources/features/schema.yml (hunk 2 above)
Comment (gabrielThiffault) [reply models/marts/human_resources/features/schema.yml:5]: Nope

File: models/marts/human_resources/features/schema.yml (hunk 3 above)
Comment (gabrielThiffault) [reply models/marts/human_resources/features/schema.yml:44]: Retiré

File: models/marts/human_resources/features/active/schema.yml (hunk 4 above)
Comment (gabrielThiffault) [reply models/marts/human_resources/features/active/schema.yml:12]: Retiré

**Comment by sadqimo** [comment]
> Ça sort le même nombre d'employés, mais oui t'as raison 

je crois qu'il faut utiliser des inner dans ce cas

**Comment by sadqimo** [comment]
> Oui, sans les règles d'affaires nous arrivons à 2269 et le script lui donne 2231 

les RH de votre coté sont correcte avec ce chiffre ?

**Comment by gabrielThiffault** [comment]
> @gabrielThiffault, peux tu m'explique stp la différence entre la table " emp_actif_fact_base", la "fact_activity_current" et la "fact_today_active"

1. fact_activity_current: retiré
2. fact_activity_current : Donne uniquement  les employés actuellement actifs "where last_pay_date > dateadd(week, -2, getdate())"
3. emp_actif_fact_base : Permet de joindre les deux, employés actuellement actifs et actif

**Comment by gabrielThiffault** [comment]
> > Oui, sans les règles d'affaires nous arrivons à 2269 et le script lui donne 2231
> 
> les RH de votre coté sont correcte avec ce chiffre ?

Oui

File: models/dashboards/emp_actif/features/emp_actif_fact_emp_actif.sql [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#}
+{{ config(alias="emp_actif_fact_emp_actif") }}
+
+select
+    act.matr,
+    act.type as 'asdf34',
```
Comment (sadqimo) [inline models/dashboards/emp_actif/features/emp_actif_fact_emp_actif.sql:22]: @gabrielThiffault, ca veut dire quoi asdf34 ?

File: models/dashboards/emp_actif/features/emp_actif_fact_emp_actif.sql [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+    emp.legal_name,
+    emp.sex as gender,
```
Comment (sadqimo) [inline models/dashboards/emp_actif/features/emp_actif_fact_emp_actif.sql:24]: @gabrielThiffault à mofier suivant les modifications du nom de la colonne dans la table dim_employees

File: models/dashboards/emp_actif/features/emp_actif_fact_emp_actif.sql [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+        then 'Temporaire (paie automatique)'
+        when stat.is_reg = 0 and act.type = 'P'
+        then 'Temporaire (sur pièce)'
+        when stat.is_reg = 1
+        then 'Régulier'
+        else 'Autres'
+    end as 'type'
+from {{ ref("emp_actif_fact_base") }} as act
+inner join {{ ref("dim_employees") }} as emp on act.matr = emp.matr
+left join {{ ref("dim_mapper_workplace") }} lieu on act.workplace = lieu.workplace
```
Comment (sadqimo) [inline models/dashboards/emp_actif/features/emp_actif_fact_emp_actif.sql:59]: @gabrielThiffault est ce qu'on peut remplacer les left join par des inner? (c'est une question dont je comnnais pas la réponse :) )

**Comment by sadqim** [comment]
@gabrielThiffault suggestion : ajouter les étiquettes de données
Capture d’écran 2024-04-02 085216

**Comment by sadqim** [comment]
@gabrielThiffault est ce que ca correspond au nombre d'employés ? si oui il serait mieux de le mettre dans un encadrer avec un titre  
image

**Comment by sadqim** [comment]
@gabrielThiffault  autre suggestion mettre une place dans le tdb powerbi pour mettre un lien vers le rapport reportbuilder. pour que chacun insert le lien quand il aura publier le tdb

File: models/dashboards/emp_actif/features/emp_actif_fact_emp_actif.sql (hunk 7 above)
Comment (gabrielThiffault) [reply models/dashboards/emp_actif/features/emp_actif_fact_emp_actif.sql:22]: Fichue de bonne question :), probablement des tests. Il a été supprimé

File: models/dashboards/emp_actif/features/emp_actif_fact_emp_actif.sql [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+        when act.mode_cour = 'A'
+        then 'Annuel'
+        when act.mode_cour = 'H' or act.mode_cour = 'L'
+        then 'Heure'
+        when act.mode_cour = 'E'
+        then 'Enseigant(e) - 10 mois'
+        when act.mode_cour = '1'
+        then 'Suppléant(e)'
+        else 'Autres'
+    end as 'renumeration_mode',
```
[the remark this answers is not in the document]
Comment (gabrielThiffault) [reply models/dashboards/emp_actif/features/emp_actif_fact_emp_actif.sql:42]: De mémoire, oui, je me renseigne demain auprès de Mélanie.

File: models/dashboards/emp_actif/features/emp_actif_fact_emp_actif.sql (hunk 9 above)
Comment (gabrielThiffault) [reply models/dashboards/emp_actif/features/emp_actif_fact_emp_actif.sql:59]: Pour les lefts join présents, ont ne peut pas les remplacer par des inner join

File: models/dashboards/emp_actif/features/emp_actif_fact_emp_actif.sql (hunk 10 above)
[the remark this answers is not in the document]
Comment (gabrielThiffault) [reply models/dashboards/emp_actif/features/emp_actif_fact_emp_actif.sql:41]: **Le mode de rémunération est régi par les ententes nationales donc tous les CSS doivent suivre le mode.

**Comment by gabrielThiffault** [comment]
**Tableau report builder mis à jour :** 
1. L'impression fonctionne sans saut de page inutile. Pour y arriver, nous avons dû retirer la colonne "Adresse courriel"
2. L'en-tête s'affiche maintenant sur toutes les pages

**Comment by sadqim** [comment]
@gabrielThiffault , possible d'agrandir les plages des paramètres 
image

**Comment by gabrielThiffault** [comment]
> @gabrielThiffault , possible d'agrandir les plages des paramètres image

Non ce n'est pas possible, j'ai regardé avec Fred et nous n'avons rien trouvé

**Review by sadqim — APPROVED** [review]
LGTM

**Comment by jalluard** [comment]
Il y a 2 coquilles dans ta section "" : 
- Remplacer `cd ../<cssXX>.dashboards_store` par `cd ../<cssXX>.data.tbe`
- Remplacer `git checkout feature/feature/rh_emp_actif` par `git checkout feature/rh_emp_actif`

File: models/marts/human_resources/features/activity/fact_activity_current.sql (hunk 5 above)
Comment (jalluard) [reply models/marts/human_resources/features/activity/fact_activity_current.sql:38]: Mettre le prefixe dme avant last_pay_date

File: models/marts/human_resources/features/schema.yml [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+      Table qui enregistre l'ensemble des employés actifs avec l'état de leur permanence. 
+      Il est entendu ici que la permanence est une période de deux ans, soit à partir de la date de début du poste actuel 
+      ou d'un poste dans le même groupe d'emploi et la date courrante et ce sans cessation d'emploi.
+    tests:
+      - resolution:
+          combination_of_columns:
+            - matricule
+            - corps_emploi
+            - permanence
+
```
Comment (jalluard) [inline models/marts/human_resources/features/schema.yml:19]: Souci ici non?

**Review by jalluard — CHANGES_REQUESTED** [review]
juste corriger les choses mentionnées dans mes 2 commentaires. Belle PR ;)

**Comment by sadqim** [comment]
allo @gabrielThiffault est ce que c'est réglé ?

**Comment by sadqim** [comment]
> allo @gabrielThiffault est ce que c'est réglé ?

Salut @gabrielThiffault est ce qu'il y a bloquant de ton coté ?

File: models/marts/human_resources/features/schema.yml (hunk 11 above)
Comment (gabrielThiffault) [reply models/marts/human_resources/features/schema.yml]: Effectivement, il y a avait de l'information qui se contredisait

**Review by sadqim — APPROVED** [review]

**Review by jalluard — APPROVED** [review]

**Pull request merged**